rfc/halyard-lite.md

```diff
@@ -0,0 +1,475 @@
+so that users using downstream tools (Operator, Helm, Kustomize) can easily get
+the container that should be deployed for each microservice without needing to
+resolve the BOM themselves.
+
+### Install Pathways
+
+With the new Halyard Lite, there will be a number of install pathways available
+to users.
+
+#### Operator
+
+The operator being built by Armory will replace its dependency on the existing
+Halyard with Halyard Lite.  As the operator already abstracts away the parts of
+Halyard that are being removed, this change should be transparent to the end user.
+
+#### Kustomize
+
+We’ll provide a Kustomize repo that contains canonical examples of the Kubernetes
+YAML for deploying Spinnaker.  In order to generate a full deployment, users will
+first run Halyard Lite to generate their config files, then will point Kustomize
+at the output directory containing these files so that it can generate the
+required ConfigMaps as well as overrides for the docker container versions.
+
+#### Helm
+
+The solution here is less clear, as there is not a great way to pass config files
+to Helm; to have a native Helm installation, it would be necessary to reproduce
+the entire Halyard config in the values file, and re-implement Halyard Lite in
+Helm’s templating language.  A better solution would be to have Helm install the
+operator, which is similar to how the chart currently just installs Halyard but
+should provide a more Kubernetes-native experience.
+
+#### Manual
+
+Users with enough Kubernetes experience, or with enough special use cases may
+want to manually write the YAML for their Spinnaker deployment.  In this case, they’d
+use Halyard Lite to generate the required service configuration files, and would
+then feed these into their process for generating the YAML required to deploy Spinnaker.
+
+### Technical Overview
+
+#### Refactor or replace
+
+Given the above discussion about the shortcomings of the current implementation
+of Halyard, and the fact that we are proposing to only support a small core of
+the current functionality, we will re-implement this core part instead of
+refactoring Halyard.
+
+This has the advantage of allowing the existing Halyard to be maintained while
+the new “Halyard Lite” is being implemented and adopted by end users; it also
+insulates the existing Halyard from quick iteration on Halyard Lite. One disadvantage
+of this approach is that there will be a period of time where we need to maintain
+both the old and new Halyard.
+
+#### CLI vs. Daemon
+
+Halyard is currently implemented as a CLI and a daemon, greatly adding to its
+complexity.  This decision was made to allow support for clients other than the
+CLI, but to date no such implementations have been written and none are planned.
+The new tool will replace this with a CLI only, with no background daemon process.
+
+#### Language
+
+Halyard Lite will be written in Go, which has the following advantages:
```

Comment:
+100 to be written in Go - I would also strongly suggest code authors and reviewers be well versed in idiomatic Go, as it sometimes has quirks not present in other languages. (cc @brandonnelson3)

Reply:
Yes, we definitely plan on making sure to learn Go's idioms to avoid writing Java that has been lightly translated to Go.  We're also working with @ncknt who's written a lot of Go (including the Armory operator).

Reply:
:100: Another advantage is that you don't need to spin up the JVM to run some simple commands. That lag is infuriating for command-line tooling.